src/main/java/picard/cmdline/CommandLineProgram.java

```diff
@@ -347,6 +335,19 @@ protected boolean parseArgs(final String[] argv) {
         }
         REFERENCE_SEQUENCE = referenceSequence.getReferenceFile();
 
+        // Provide one temp directory if the caller didn't
+        if (this.TMP_DIR == null) this.TMP_DIR = new ArrayList<>();
```

Comment:
I'm worried that moving this chunk of code down will break other things.  If you look above we're instantiating the intel libraries which requires a tmp dir.   I would probably put this as close to the top as possible.  Also, it should really have a comment talking about the timing issues so we don't repeat this problem.  

It wouldn't be crazy to extract this as a method `setTempDirectory` or something along those lines.